# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../strace/arch-1-avr32-add-syscalls-up-to-sysvipc.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Index: strace-4.5.14/linux/avr32/syscallent.h
===================================================================
--- strace-4.5.14.orig/linux/avr32/syscallent.h	2007-02-19 07:48:12.000000000 +0100
+++ strace-4.5.14/linux/avr32/syscallent.h	2007-02-19 08:01:24.000000000 +0100
@@ -251,4 +251,57 @@
 	{ 3,	TS,	sys_tgkill,		"tgkill"	}, /* 221 */
 	{ 5,	0,	printargs,		"SYS_222"	}, /* 222 */
 	{ 2,	TF,	sys_utimes,		"utimes"	}, /* 223 */
-	{ 6,	0,	printargs,		"fadvise64_64"	}, /* 224 */
+	{ 6,	0,	sys_fadvise64_64,	"fadvise64_64"	}, /* 224 */
+	{ 3,	0,	printargs,		"cacheflush"	}, /* 225 */
+	{ 0,	0,	printargs,		"vserver"	}, /* 226 */
+	{ 4,	0,	sys_mq_open,		"mq_open"	}, /* 227 */
+	{ 1,	0,	sys_mq_unlink,		"mq_unlink"	}, /* 228 */
+	{ 5,	0,	sys_mq_timedsend,	"mq_timedsend"	}, /* 229 */
+	{ 5,	0,	sys_mq_timedreceive,	"mq_timedreceive" }, /* 230 */
+	{ 2,	0,	sys_mq_notify,		"mq_notify"	}, /* 231 */
+	{ 3,	0,	sys_mq_getsetattr,	"mq_getsetattr" }, /* 232 */
+	{ 4,	0,	printargs,		"kexec_load"	}, /* 233 */
+	{ 1,	0,	sys_waitid,		"waitid"	}, /* 234 */
+	{ 5,	0,	printargs,		"add_key"	}, /* 235 */
+	{ 4,	0,	printargs,		"request_key"	}, /* 236 */
+	{ 5,	0,	printargs,		"keyctl"	}, /* 237 */
+	{ 3,	0,	printargs,		"ioprio_set"	}, /* 238 */
+	{ 2,	0,	printargs,		"ioprio_get"	}, /* 239 */
+	{ 1,	0,	printargs,		"inotify_init"	}, /* 240 */
+	{ 3,	0,	printargs,		"inotify_add_watch" }, /* 241 */
+	{ 2,	0,	printargs,		"inotify_rm_watch" }, /* 242 */
+	{ 4,	0,	printargs,		"openat"	}, /* 243 */
+	{ 3,	0,	printargs,		"mkdirat"	}, /* 244 */
+	{ 4,	0,	printargs,		"mknodat"	}, /* 245 */
+	{ 5,	0,	printargs,		"fchownat"	}, /* 246 */
+	{ 3,	0,	printargs,		"futimesat"	}, /* 247 */
+	{ 4,	0,	sys_fstatfs64,		"fstatat64"	}, /* 248 */
+	{ 3,	0,	printargs,		"unlinkat"	}, /* 249 */
+	{ 4,	0,	printargs,		"renameat"	}, /* 250 */
+	{ 5,	0,	printargs,		"linkat"	}, /* 251 */
+	{ 3,	0,	printargs,		"symlinkat"	}, /* 252 */
+	{ 4,	0,	printargs,		"readlinkat"	}, /* 253 */
+	{ 3,	0,	printargs,		"fchmodat"	}, /* 254 */
+	{ 3,	0,	printargs,		"faccessat"	}, /* 255 */
+	{ 6,	0,	printargs,		"pselect6"	}, /* 256 */
+	{ 5,	0,	printargs,		"ppoll"		}, /* 257 */
+	{ 1,	0,	printargs,		"unshare"	}, /* 258 */
+	{ 2,	0,	printargs,		"set_robust_list" }, /* 259 */
+	{ 3,	0,	printargs,		"get_robust_list" }, /* 260 */
+	{ 6,	0,	printargs,		"splice"	}, /* 261 */
+	{ 4,	0,	printargs,		"sync_file_range" }, /* 262 */
+	{ 4,	0,	printargs,		"tee"		}, /* 263 */
+	{ 4,	0,	printargs,		"vmsplice"	}, /* 264 */
+	{ 6	0,	printargs,		"sys_epoll_pwait" }, /* 265 */
+	{ 2,	TI,	sys_msgget,		"msgget"	}, /* 266 */
+	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 267 */
+	{ 5,	TI,	sys_msgrcv,		"msgrcv"	}, /* 268 */
+	{ 3,	TI,	sys_msgctl,		"msgctl"	}, /* 269 */
+	{ 3,	TI,	sys_semget,		"semget"	}, /* 270 */
+	{ 3,	TI,	sys_semop,		"semop"		}, /* 271 */
+	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 272 */
+	{ 4,	TI,	sys_semtimedop,		"semtimedop"	}, /* 273 */
+	{ 3,	TI,	sys_shmat,		"shmat"		}, /* 274 */
+	{ 3,	TI,	sys_shmget,		"shmget"	}, /* 275 */
+	{ 1,	TI,	sys_shmdt,		"shmdt"		}, /* 276 */
+	{ 3,	TI,	sys_shmctl,		"shmctl"	}, /* 277 */
Index: strace-4.5.14/system.c
===================================================================
--- strace-4.5.14.orig/system.c	2007-02-19 07:48:12.000000000 +0100
+++ strace-4.5.14/system.c	2007-02-19 07:48:12.000000000 +0100
@@ -1578,7 +1578,6 @@
 	{ CTL_KERN, "CTL_KERN" },
 	{ CTL_VM, "CTL_VM" },
 	{ CTL_NET, "CTL_NET" },
-	/* was CTL_PROC */
 	{ CTL_FS, "CTL_FS" },
 	{ CTL_DEBUG, "CTL_DEBUG" },
 	{ CTL_DEV, "CTL_DEV" },
